src/repositorios/volatil/repositorio_materias_volatil.py

```diff
@@ -11,3 +13,20 @@ def __init__(self):
 
     def getAllMaterias(self) -> list:
         return self.materias
+
+    def getMateriaPorID(self, id: str) -> object:
+        materia = None
+        for m in self.materias:
```

Comment:
Da uma fuçada em `one-liners`. São abstracoes poderosas, pra deixar o codigo mais enxuto: [link](https://wiki.python.org/moin/Powerful%20Python%20One-Liners)

Nesse caso: `materiasEncontradas = [materia for materia in self.materias if materia.codigo == CodigoDisciplina[id.upper()]`